### PR TITLE
fix(transformer-directives): vue-scoped generate css code

### PR DESCRIPTION
**Pull request by Simon-He95**
close: #4021

image

### CODE REVIEW

File: packages/transformer-directives/src/apply.ts [hunk 1]
```diff
@@ -83,7 +84,8 @@ export async function parseApply({ code, uno, applyVariable }: TransformerDirect
     if (parent || (selectorOrGroup && selectorOrGroup !== '.\\-')) {
       let newSelector = generate(node.prelude)
       const className = code.slice(node.prelude.loc!.start.offset, node.prelude.loc!.end.offset)
-      if (selectorOrGroup && selectorOrGroup !== '.\\-') {
+      // if mode is 'vue-scoped', we don't need to scope the selector
+      if ((uno.userConfig as VitePluginConfig).mode !== 'vue-scoped' && selectorOrGroup && selectorOrGroup !== '.\\-') {
```
Comment (zyyv) [inline packages/transformer-directives/src/apply.ts:88]: This `mode` property is not on `userConfig`

Comment (Simon-He95) [reply packages/transformer-directives/src/apply.ts:88]: It seems that there are some problems with the type, but it does exist at runtime. It should be merged by the userConfig from vitePlugin.

Comment (zyyv) [reply packages/transformer-directives/src/apply.ts:88]: https://github.com/unocss/unocss/blob/243623ddb7decd7002e4110ef3674e1feeef6297/packages/config/src/index.ts#L59-L60

It seems that when loading the config, the vite plugin config is not merged into the `userConfig`

Comment (Simon-He95) [reply packages/transformer-directives/src/apply.ts:88]: https://github.com/unocss/unocss/blob/main/packages/vite/src/index.ts#L35
Go down from here and there is a merge operation

image

image

Comment (zyyv) [reply packages/transformer-directives/src/apply.ts:88]: image
image

vite plugin config is **not** injected into the `context`.

Comment (Simon-He95) [reply packages/transformer-directives/src/apply.ts:88]: https://github.com/unocss/unocss/blob/main/packages/vite/src/index.ts#L48
`ConfigHMRPlugin` will trigger reloadConfig and then inject
image

Comment (zyyv) [reply packages/transformer-directives/src/apply.ts:88]: I think `reloadConfig` still won't inject vite config in `userConfig`, it is a configuration item of `unocss` not `vite` plugin option. 🧐

Comment (Simon-He95) [reply packages/transformer-directives/src/apply.ts:88]: image

image
If you debug, you can see that userConfig already has `mode: vue-scoped` before transformerDirectives are executed.

**Comment by zyyv** [comment]
I will finish the review today, give me some time. ❤️🥹

**Comment by zyyv** [comment]
@Simon-He95 I'll close this pr because it has been resolved upstream.

**Pull request closed without merging**